runtime/asm/start_prototype.rs

```diff
@@ -31,32 +31,33 @@ extern fn start_prototype(
     _app_break: usize,
 ) -> ! {
     use crate::TockSyscalls;
-    use libtock_platform::{OneArgMemop, RawSyscalls, YieldType};
+    use libtock_platform::RawSyscalls;
 
     let pc: usize;
-    #[cfg(target_arch = "riscv32")]
     unsafe {
+        #[cfg(target_arch = "arm")]
+        asm!("mov {}, pc", lateout(reg) pc, options(nomem, nostack, preserves_flags));
+        #[cfg(target_arch = "riscv32")]
         asm!("auipc {}, 0", lateout(reg) pc, options(nomem, nostack, preserves_flags));
     }
     if pc != rt_header.start {
         // Binary is in an incorrect location: report an error via
-        // LowLevelDebug.
+        // LowLevelDebug then exit.
         unsafe {
-            TockSyscalls::four_arg_syscall(8, 1, 2, 0, 2);
-        }
-        // TODO: Replace with an Exit call when exit is implemented.
-        loop {
-            TockSyscalls::raw_yield(YieldType::Wait);
+            TockSyscalls::syscall4::<2>([8 as *mut (), 1 as *mut (), 2 as *mut (), 0 as *mut ()]);
+            TockSyscalls::syscall2::<6>([0 as *mut (), 0 as *mut ()]);
```

Comment:
This would be a little clearer if it used named constants for the generic parameters

Reply:
Done. I ran `rustfmt` on the file which produced a few unrelated changes too (CI doesn't check this file's formatting).